Firestore/Source/Public/FirebaseFirestore/FIRQuery.h

```diff
@@ -308,6 +332,36 @@ NS_SWIFT_NAME(Query)
 - (FIRQuery *)queryWhereFieldPath:(FIRFieldPath *)path
                                in:(NSArray<id> *)values NS_SWIFT_NAME(whereField(_:in:));
 
+/**
+ * Creates and returns a new `FIRQuery` with the additional filter that documents must contain
+ * the specified field and the value does not equal any of the values from the provided array.
+ *
+ * A query can have only one `notIn` filter, and it cannot be combined with an `arrayContains`,
+ * `arrayContainsAny`, `in`, or `notEqual` filter.
```

Comment:
Is it worth calling out the behavior of not-in [null] here? It's counter-intuitive because not-equal null automagically makes that work, but not-in has no equivalent.

Reply:
Yeah, this is tough. Here's my take, which avoids "results in no document matches" which is a fairly indirect way of saying that the filter can't match the field.

> One special case is that `notIn` filters cannot match `nil` values. To query for documents where a field exists and is `nil`, use a `notEqual` filter, which can handle this special case.

Separately, note that Objective-C and Swift spell "null" as `nil`.